t/mojo/reactor_ev.t

```diff
@@ -8,307 +8,340 @@ plan skip_all => 'EV 4.32+ required for this test!'                  unless eval
-$reactor->timer(1 => sub { shift->stop });
-$reactor->start;
-ok $readable,  'handle is readable';
-ok !$writable, 'handle is not writable';
+subtest 'Listen' => sub {
+  my $listen = IO::Socket::INET->new(Listen => 5, LocalAddr => '127.0.0.1');
+  my ($readable, $writable);
+  my $reactor = Mojo::Reactor::EV->new;
+  $reactor->io($listen => sub { pop() ? $writable++ : $readable++ })->watch($listen, 0, 0)->watch($listen, 1, 1);
+  $reactor->timer(0.025 => sub { shift->stop });
+  $reactor->start;
+  ok !$readable,                 'handle is not readable';
+  ok !$writable,                 'handle is not writable';
+  ok $reactor->remove($listen),  'removed';
+  ok !$reactor->remove($listen), 'not removed again';
+};
 
-# Accept
-my $server = $listen->accept;
-ok $reactor->remove($listen),  'removed';
-ok !$reactor->remove($listen), 'not removed again';
-($readable, $writable) = ();
-$reactor->io($client => sub { pop() ? $writable++ : $readable++ });
-$reactor->again($reactor->timer(0.025 => sub { shift->stop }));
-$reactor->start;
-ok !$readable, 'handle is not readable';
-ok $writable,  'handle is writable';
-print $client "hello!\n";
-sleep 1;
-ok $reactor->remove($client), 'removed';
-($readable, $writable) = ();
-$reactor->io($server => sub { pop() ? $writable++ : $readable++ });
-$reactor->watch($server, 1, 0);
-$reactor->timer(0.025 => sub { shift->stop });
-$reactor->start;
-ok $readable,  'handle is readable';
-ok !$writable, 'handle is not writable';
-($readable, $writable) = ();
-$reactor->watch($server, 1, 1);
-$reactor->timer(0.025 => sub { shift->stop });
-$reactor->start;
-ok $readable, 'handle is readable';
-ok $writable, 'handle is writable';
-($readable, $writable) = ();
-$reactor->watch($server, 0, 0);
-$reactor->timer(0.025 => sub { shift->stop });
-$reactor->start;
-ok !$readable, 'handle is not readable';
-ok !$writable, 'handle is not writable';
-($readable, $writable) = ();
-$reactor->watch($server, 1, 0);
-$reactor->timer(0.025 => sub { shift->stop });
-$reactor->start;
-ok $readable,  'handle is readable';
-ok !$writable, 'handle is not writable';
-($readable, $writable) = ();
-$reactor->io($server => sub { pop() ? $writable++ : $readable++ });
-$reactor->timer(0.025 => sub { shift->stop });
-$reactor->start;
-ok $readable, 'handle is readable';
-ok $writable, 'handle is writable';
+subtest 'Interdependent tests' => sub {
+  my $reactor = Mojo::Reactor::EV->new;
+  my $listen  = IO::Socket::INET->new(Listen => 5, LocalAddr => '127.0.0.1');
+  my $port    = $listen->sockport;
+  my ($readable, $writable);
+  my ($timer,    $recurring);
+  my ($single,   $pair, $one, $two, $last);
+  my $client   = IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => $port);
+  my $reactor2 = Mojo::Reactor::EV->new;
+  my $id       = $reactor->recurring(0 => sub { $recurring++ });
+  my $server;
+  $reactor->io($listen => sub { pop() ? $writable++ : $readable++ })->watch($listen, 0, 0)->watch($listen, 1, 1);
```

Comment:
The fact that not all of these added lines have existed in the removed lines below, makes me believe that this might be a little more verbose than it has to be.

Reply:
If you want I can make the previously global objects global again since I did not achieve to make every test independent between themselves anyway.

Reply:
Might be best. Whatever results in the simplest code.

Reply:
I think it is now better.